src/parser/postgresparser.cpp

```diff
@@ -800,6 +800,143 @@ parser::SQLStatement* PostgresParser::CreateTransform(CreateStmt* root) {
   return reinterpret_cast<parser::SQLStatement*>(result);
 }
 
+// This helper function takes in a Postgres FunctionParameter object and transforms
+// it into a Peloton FunctionParameter object
+parser::FuncParameter* PostgresParser::FunctionParameterTransform(FunctionParameter* root) {
+  parser::FuncParameter::DataType data_type;
+  TypeName* type_name = root->argType;
+  char* name = (reinterpret_cast<value*>(type_name->names->tail->data.ptr_value)
+                    ->val.str);
+  parser::FuncParameter* result = nullptr;
+
+  // Transform parameter type
+  if ((strcmp(name, "int") == 0) || (strcmp(name, "int4") == 0)) {
+    data_type = FuncParameter::DataType::INT;
+  } 
+  //*******See which of the below are needed later***********
+  // I think text/varchar may be used for the pg_catalog
+  else if (strcmp(name, "varchar") == 0) {
+    data_type = FuncParameter::DataType::VARCHAR;
+  } else if (strcmp(name, "int8") == 0) {
+    data_type = FuncParameter::DataType::BIGINT;
+  } else if (strcmp(name, "int2") == 0) {
+    data_type = FuncParameter::DataType::SMALLINT;
+  } else if (strcmp(name, "text") == 0) {
+    data_type = FuncParameter::DataType::TEXT;
+  } else if (strcmp(name, "tinyint") == 0) {
+    data_type = FuncParameter::DataType::TINYINT;
+  } else if(strcmp(name, "bool") == 0) {
+     data_type = FuncParameter::DataType::BOOL;
+  } else {
+    LOG_ERROR("Column DataType %s not supported yet...\n", name);
+    throw NotImplementedException("...");
```

Comment:
I think it would be more useful if the Exception message returned the same information as is logged in the ERROR stream.